ddpui/api/org_preferences_api.py

```diff
@@ -0,0 +1,306 @@
+
+    # Structure the preferences response
+    preferences = {
+        "org": {
+            "id": org_preferences.org.id,
+            "name": org_preferences.org.name,
+            "slug": org_preferences.org.slug,
+            "type": org_preferences.org.type,
+        },
+        "trial_start_date": org_preferences.trial_start_date,
+        "trial_end_date": org_preferences.trial_end_date,
+        "llm_optin": org_preferences.llm_optin,
+        "llm_optin_approved_by": approved_by_data,
+        "llm_optin_date": org_preferences.llm_optin_date,
+        "enable_discord_notifications": org_preferences.enable_discord_notifications,
+        "discord_webhook": org_preferences.discord_webhook,
+    }
+
+    return {"success": True, "res": preferences}
+
+
+@orgpreference_router.get("/org-superset", auth=auth.CustomAuthMiddleware())
+def get_org_superset_details(request):
+    """Gets preferences for an organization by org_id"""
+    orguser: OrgUser = request.orguser
+    org = orguser.org
+
+    try:
+        org_superset = OrgSupersets.objects.select_related(
+            "org",
+        ).get(org=org)
+
+    except OrgSupersets.DoesNotExist:
+        raise HttpError(404, "Organizations superset details not found")
+
+    response_data = {
+        "org": {
+            "id": org_superset.org.id,
+            "name": org_superset.org.name,
+            "slug": org_superset.org.slug,
+            "type": org_superset.org.type,
+        },
+        "superset_version": org_superset.superset_version,
+    }
+    return {"success": True, "res": response_data}
+
+
+@orgpreference_router.post("/org-superset", auth=auth.CustomAuthMiddleware())
+def create_org_superset_details(request, payload: CreateOrgSupersetDetailsSchema):
+    """Creates or updates an entry in db with org superset details."""
+    orguser: OrgUser = request.orguser
+    org = orguser.org
+
+    org_superset, created = OrgSupersets.objects.update_or_create(org=org, defaults=payload.dict())
+
+    response_data = {
+        "org": {
+            "name": org.name,
+            "slug": org.slug,
+            "type": org.type,
+        },
+        "superset_version": org_superset.superset_version,
+        "created": created,
+    }
+
+    return {"success": True, "res": response_data}
+
+
+@orgpreference_router.get("/toolinfo", auth=auth.CustomAuthMiddleware())
+def get_tools_versions(request):
+    orguser: OrgUser = request.orguser
+    org = orguser.org
+
+    try:
+        # Validate org_id
+        org_superset = OrgSupersets.objects.get(org=org)
+    except OrgSupersets.DoesNotExist:
+        return {"success": False, "error": "Organization not found"}
+
+    versions = []
+
+    # Airbyte Version
+    try:
+        airbyte_url = AIRBYTE_URL_TO_GET_VERSION
+        airbyte_response = requests.get(airbyte_url)
+        if airbyte_response.status_code == 200:
+            airbyte_data = airbyte_response.json()
```

Comment:
we use airbyte_service to talk/call any apis of airbyte. lets move this as a helper there